lib/persist-state.c

```diff
@@ -174,6 +175,22 @@ _wait_until_map_release(PersistState* self)
 }
 
 static gboolean
+_increase_file_size(PersistState *self, guint32 new_size)
+{
+  ssize_t length = new_size - self->current_size;
+  gchar *pad_buffer = g_new0(gchar, length);
+  gboolean result = pwrite_strict(self->fd, pad_buffer, length, self->current_size);
+  if (!result)
+    {
+      msg_error("Can't grow the persist file", evt_tag_int("old_size", self->current_size), evt_tag_int("new_size", new_size), evt_tag_errno("error", errno), NULL);
+    }
+  g_free(pad_buffer);
+  return result;
+}
```

Comment:
why isn't the lseek + write a single byte enough to extend the file? this is documented to be working in POSIX systems, but I think it works the same on Windows as well. is there a specific reason?

the msg_error() line should be broken down, with all tag-value pairs in a separate lines.